androidHackathonApp/src/main/java/com/kcc/kmmhackathon/androidHackathonApp/MealsAdapter.kt

```diff
@@ -0,0 +1,77 @@
+package com.kcc.kmmhackathon.androidHackathonApp
+
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.TextView
+import androidx.core.content.ContextCompat
+import androidx.recyclerview.widget.RecyclerView
+import com.kcc.kmmhackathon.shared.entity.Meal
+import java.time.LocalDate
+import kotlin.math.exp
+
+class MealsAdapter (var mealsList: List<Meal>) : RecyclerView.Adapter<MealsAdapter.MealsViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MealsViewHolder {
+        return LayoutInflater.from(parent.context)
+            .inflate(R.layout.meal_item_row, parent, false)
+            .run(::MealsViewHolder)
+    }
+
+    override fun getItemCount(): Int {
+        return mealsList.size
+    }
+
+    override fun onBindViewHolder(holder: MealsViewHolder, position: Int) {
+        holder.bindData(mealsList[position])
+    }
+
+
+    inner class MealsViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+        private val nameView = itemView.findViewById<TextView>(R.id.mealName)
+        private val tempView = itemView.findViewById<TextView>(R.id.mealTemp)
+        private val infoView = itemView.findViewById<TextView>(R.id.mealInfo)
+        private val availableView = itemView.findViewById<TextView>(R.id.mealAvailable)
+        private val expiryView = itemView.findViewById<TextView>(R.id.mealExpiry)
+        private val portionsView = itemView.findViewById<TextView>(R.id.mealPortions)
+        private val reserveButton = itemView.findViewById<Button>(R.id.reserveButton)
+
+        fun bindData(meal: Meal) {
+            val ctx = itemView.context
+            nameView.setText(meal.name)
+            tempView.setText(if (meal.hot) "Hot" else "Cold")
```

Comment:
the tempView text is seeming to be set twice?

Reply:
Would it be easier to have a `temp` var so we could have `tempView.text == meal.temp` ?

Reply:
As far as I know you don't need to `setText` you can `view.text` and it will apply for setting the text